lib/EscriptGrammar/EscriptLexer.g4

```diff
@@ -82,18 +82,16 @@ TOK_IN:             'in';
 
 // Literals
 
-DECIMAL_LITERAL:    ('0' | [1-9] (Digits? | '_'+ Digits)) [lL]?;
-HEX_LITERAL:        '0' [xX] [0-9a-fA-F] ([0-9a-fA-F_]* [0-9a-fA-F])? [lL]?;
-OCT_LITERAL:        '0' '_'* [0-7] ([0-7_]* [0-7])? [lL]?;
-BINARY_LITERAL:     '0' [bB] [01] ([01_]* [01])? [lL]?;
+DECIMAL_LITERAL:    ('0' | [1-9] Digits?);
+HEX_LITERAL:        '0' [xX] HexDigits;
+OCT_LITERAL:        '0' [0-7]+;
+BINARY_LITERAL:     '0' [bB] [01]+;
                     
-FLOAT_LITERAL:      (Digits '.' Digits? | '.' Digits) ExponentPart? [fFdD]?
-             |       Digits (ExponentPart [fFdD]? | [fFdD])
+FLOAT_LITERAL:      (Digits '.' Digits? | '.' Digits) ExponentPart?
+             |       Digits ExponentPart
              ;
 
-HEX_FLOAT_LITERAL:  '0' [xX] (HexDigits '.'? | HexDigits? '.' HexDigits) [pP] [+-]? Digits [fFdD]?;
-
-CHAR_LITERAL:       '\'' (~['\\\r\n] | EscapeSequence) '\'';
+HEX_FLOAT_LITERAL:  '0' [xX] (HexDigits '.'? | HexDigits? '.' HexDigits);
```

Comment:
This is missing the exponent part of the hex float. Do we want to leave it out?

Reply:
You are right. When attempting to compile `0x10.1` in cpp,  you get an error:
```
test.cc:4:29: error: hexadecimal floating literal requires an exponent
    cout << "Hel " << 0x10.1;
```

So the `p` is required.